### DISCUSSION TITLE
How do I add PDF.js to a Next.js project?

### DISCUSSION THREAD

**Discussion by courtneypattison** [vercel/next.js#60775] [Help]
### Summary

I'm having trouble adding PDF.js to a Next.js project. I've tried setting up multiple entries in the webpack config as per [Mozilla's webpack example](https://github.com/mozilla/pdf.js/blob/master/examples/webpack/webpack.config.js). However, when I do something like that, my dev build crashes. See example for commented out attempt. I've tried experimental urlImports (see example) but that gives me the runtime error `Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.`. I've even tried copying the `pdf.worker.min.mjs` file in a postinstall script using `fs.copyFileSync()` but I can't get it to run with the `next dev` script.

### Example

https://codesandbox.io/p/devbox/elegant-mccarthy-y6v9yk?file=%2Fapp%2Fpage.tsx

**Comment by mmo80** [comment]
Maybe this helps: [https://stackoverflow.com/a/78121976/99769](https://stackoverflow.com/a/78121976/99769)

**Comment by Cygra** [comment]
The @ungap/with-resolvers solution works for me

https://stackoverflow.com/a/79099873/9217735
